docket/plugins/config_scripts.py

```diff
@@ -44,9 +43,12 @@ def predicate(message: hikari.GuildMessageCreateEvent) -> bool:
         attachment = msg.message.attachments[0]
         content = str((await attachment.read()).decode("utf-8"))
     elif not msg.message.attachments:
-        match = CODE_BLOCK.search(msg.message.content)
-        assert match
-        content = match.group("code").strip("`")
+        lines = msg.message.content.split("\n")
+        if lines[0].strip() in {"```", "```lua"}:
+            lines.pop(0)
+        if lines[-1].strip() == "```":
+            lines.pop(-1)
+        content = "\n".join(lines)
```

Comment:
I feel like this section can be its own function
```python
def get_message_lua(self, msg: hikari.Message) -> str:
    if not msg.message.content:
        attachment = msg.message.attachments[0]
        return str((await attachment.read()).decode("utf-8"))
    elif not msg.message.attachments:
        match = CODE_BLOCK.search(msg.message.content)
        assert match
        content = match.group("code").strip("`")
        lines = msg.message.content.split("\n")
        if lines[0].strip() in {"```", "```lua"}:
            lines.pop(0)
        if lines[-1].strip() == "```":
            lines.pop(-1)
        return "\n".join(lines)

    raise DocketBaseError("You can't send both a code block and an attachment.")

Reply:
Since this is only used in one place I would opt to leave it here for now. I'll leave it up to you though.

Reply:
Code climate uses cognitive complexity which i think is a pretty good way to think about this. When your reader sees the massive if else chain, its gonna take a bit to parse what it does. A function on the other hand, instantly says its purpose because of the name, so, its much easier for the reader to understand.

Reply:
Makes sense - did this.